dotCMS/src/main/java/com/dotmarketing/factories/TreeFactory.java

```diff
@@ -212,22 +165,22 @@ public static Tree getTreeByParentAndRelationType(Inode parent, String relationT
 		return getTreesByChild(inode.getInode());
 	}
 	
-	@SuppressWarnings("unchecked")
+
 	public static List<Tree> getTreesByChild(String inode) {
 		try {
-			HibernateUtil dh = new HibernateUtil(Tree.class);
-			dh.setQuery("from tree in class com.dotmarketing.beans.Tree where child = ?");
-			dh.setParam(inode);
+			DotConnect dc = new DotConnect();
+			dc.setSQL("select * from tree where  child = ?");
+			dc.addParam(inode);
 
-			return dh.list();
+			return new DBTreeTransformer(dc.loadObjectResults()).trees();
 		} catch (Exception e) {
 			Logger.warn(TreeFactory.class, "getTree failed:" + e, e);
 		}
 
 		return new ArrayList<Tree>();
 	}
 
-	public static void swapTrees(Inode i1, Inode i2) throws HibernateException {
+	public static void swapTrees(final Inode i1, final Inode i2) throws DotDataException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.dotmarketing.exception.DotDataException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)